knowledge-base/adr/0020-generator-options.md

```diff
@@ -0,0 +1,60 @@
+# Generator (OpenAPI + OData) API
+
+## Status
+
+proposed
+
+## Context 1
+
+When generating only from one spec (either generator), which is probably the most common use case, when doing this mannually, the resulting folder structure is `<outputDir>/serviceDir`, which in turn is unexpected.
+The name of the generator command is: `generate-X-client` indicating that only one client will be generated.
+
+## Decision 1
+
+1. Different behavior if only one specification is found vs. multiple.
+   For one specification, put client directly into `<outputDir>`, for multiple make subdirectories.
+2. `generate-openapi-client batch -i <inputFile> -o <outputDir>`
+   Generate only the first matched input file found by default.
+   Show a log message in case there are multiple inputs found, that batch argument is required to nest the APIs.
+3. `generate-openapi-client -i <inputFile> -o <outputDir> --flat`
+   Flatten the directory to be `<outputDir>`.
+   Throw an error if input is a directory (with multiple files).
+4. Keep it as is. (decided)
+5. Rename the client:
+   1. `openapi-generator`
+   2. `sap-cloud-sdk-openapi-generator`
```

Comment:
@jjtang1985 @FrankEssenberger @artemkovalyov 
Please share your opinions here on what the name of the generator should be. Feel free to add other options (add them to the document directly).

Reply:
I would go for `openapi-generator` and `odata-generator` as names

Reply:
+1, as the second one is a bit too long, which is not type-friendly.

Reply:
It is decided then.